.changeset/loud-fishes-cover.md

```diff
@@ -0,0 +1,5 @@
+---
+"@atproto-labs/simple-store": minor
+---
+
+Allow `Key` type to be anything (except nullable values)
```

Comment:
I will read ahead, but could you summarize how this capability gets used?  It is certainly more complicated for implementers to support any key type, and in the past I've almost been more inclined to reign it back to just `string` instead of `string | number`.